packages/openactive-integration-tests/test/features/broker-role/no-broker/implemented/broker-included-nobroker-mode-test.js

```diff
@@ -28,31 +13,32 @@ FeatureHelper.describeFeature(module, {
   testOpportunityCriteria: 'TestOpportunityBookable',
   // The secondary opportunity criteria to use for multiple OrderItem tests
   controlOpportunityCriteria: 'TestOpportunityBookable',
+  supportsApproval: true,
 },
 function (configuration, orderItemCriteriaList, featureIsImplemented, logger) {
   describe('Booking should fail because Broker is included in Order in NoBroker mode', () => {
     describe('at C1', () => {
       // broker is included, by default, in C1 request
-      const { fetchOpportunities, c1 } = FlowStageRecipes.initialiseSimpleC1C2BFlow(orderItemCriteriaList, logger, { brokerRole: 'https://openactive.io/NoBroker' });
+      const { fetchOpportunities, c1 } = FlowStageRecipes.initialiseSimpleC1C2Flow(orderItemCriteriaList, logger, { brokerRole: 'https://openactive.io/NoBroker' });
       FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
       runFlowStageAndExpectIncompleteBrokerDetailsError(c1);
     });
     describe('at C2', () => {
       // broker is included, by default, in C2 request
-      const { fetchOpportunities, c1, c2 } = FlowStageRecipes.initialiseSimpleC1C2BFlow(orderItemCriteriaList, logger, { brokerRole: 'https://openactive.io/NoBroker', c1ReqTemplateRef: 'noBroker' });
+      const { fetchOpportunities, c1, c2 } = FlowStageRecipes.initialiseSimpleC1C2Flow(orderItemCriteriaList, logger, { brokerRole: 'https://openactive.io/NoBroker', c1ReqTemplateRef: 'noBroker' });
       FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
       FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c1);
       runFlowStageAndExpectIncompleteBrokerDetailsError(c2);
     });
-    describe('at B', () => {
-      // broker is included, by default, in B request
-      const { fetchOpportunities, c1, c2, b } = FlowStageRecipes.initialiseSimpleC1C2BFlow(orderItemCriteriaList, logger, {
+    describe('at B or P', () => {
+      // broker is included, by default, in B or P request
+      const { fetchOpportunities, c1, c2, bookRecipe } = FlowStageRecipes.initialiseSimpleC1C2BookFlow(orderItemCriteriaList, logger, {
         brokerRole: 'https://openactive.io/NoBroker', c1ReqTemplateRef: 'noBroker', c2ReqTemplateRef: 'noBroker',
       });
       FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
       FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c1);
       FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c2);
-      runFlowStageAndExpectIncompleteBrokerDetailsError(b);
+      runFlowStageAndExpectIncompleteBrokerDetailsError(bookRecipe.firstStage);
```

Comment:
An example of "expecting failure at the first stage" whether it's P or B. This is also the 1st test that I've tried that actually works in either simple or approval flow